apis/python/src/tiledbsoma/_soma_object.py

```diff
@@ -103,6 +108,10 @@ def open(
             tiledb_timestamp,
             clib_type=cls._wrapper_type._WRAPPED_TYPE.__name__,
         )
+        if _read_soma_type(handle) != cls.soma_type:
+            raise SOMAError(
+                "Unexpected SOMA metadaa encoding - unable to determine object type."
```

Comment:
I'm not sure this is the right error. Unless I'm misunderstanding this code, it looks like we were able to determine the type - it just wasn't the type we were expecting.

Reply:
You are correct - it is either a sign that we have a logic bug, OR the data was corrupted in some weird way.

I'll update the error message.